reliability/stub-core/src/main/kotlin/com/datadog/android/core/stub/StubSDKCore.kt

```diff
@@ -72,6 +72,13 @@ class StubSDKCore(
         datadogContext = datadogContext.copy(networkInfo = networkInfo)
     }
 
+    /**
+     * Returns the last metric is sent by [StubInternalLogger].
+     */
+    fun lastMetric(): Map<String, Any> {
+        return (internalLogger as StubInternalLogger).telemetryEventsWritten.last()
```

Comment:
to be precise, you need to use `filter { it["type"] == "mobile_metric" }` here, because property itself is called `telemetryEventsWritten`, so if someone else adds there non-metric telemetry, `telemetryEventsWritten.last()` will give a wrong result and things won't be correct.

Reply:
Make sense, I use `lastOrNull{ it["type"] == "mobile_metric"}` to return the last telemetry metric recorded.